### DISCUSSION TITLE
Adding preferences for multi objective problem

### DISCUSSION THREAD

**Discussion by sami-ka** [anyoptimization/pymoo#551] [Ideas]
Hi,

I wanted to know if you think that adding preferences for multi-objective optimization problem could be interesting for pymoo?
The idea would be to be a step beyond weighted sum. 
Weighted sum is a scalarizing function that defines a weight for each objective and optimizing on this criterion usually yields a single optimal solution. Going a step beyond the weighted sum here, means that the decision maker only specifies preference on the different objectives.
An example would be a ranking of the objectives : objective 1 > objective 2 > objective 3. This gives constraints for weights in the weighted sum with weight_1 >= weight _2 >= weight_3. The new "Pareto" front in this setting would be points that are not worse than any other points on a weighted sum respecting the constraint on weights.

Why is it interesting to integrate preferences? 
The pareto front will shift towards regions in the objective space that better represent the preference of the decision maker. The solving time should be improved (I have only tested for exact generation of non dominated points, so it should still be validated with meta heuristics).

How to compute it?
It is really simple! We can create a new problem being a copy of the original problem and transform the objective space of the new problem with a simple matrix multiplication. Optimizing the new problem will generate a pareto front respecting the new dominance relation defined previously. 

If you want to have more details, you can take a look at this reference: https://www.sciencedirect.com/science/article/abs/pii/S0377221717300085

Let me know what you think of this new feature. If you're interested in this, I will be able to start working on it mid February.

**Comment by blankjul** [comment]
Check out R-NSGA-III as well: https://pymoo.org/algorithms/moo/rnsga3.html

There you can provide aspiration points which will be specifically focused on during the search.
Keep in mind that normalization is key when using weights and this is usually a big challenging when redefining a single-objective optimization problem beforehand.

**Comment by sami-ka** [comment]
Thanks for your reply @blankjul, I was not aware of this approach !

It's not clear for me if you are interested in a PR on pymoo to integrate what I presented, so let me know what you think.

**Comment by blankjul** [comment]
I am always happy to merge contributions. For algorithms I prefer if either the paper is well-cited or the method has shown to outperform existing algorithms.
